src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -401,7 +399,8 @@ def delete_index(self):
 
     def query(self,
               queries: List[Query],
-              global_metadata_filter: Optional[dict] = None
+              global_metadata_filter: Optional[dict] = None,
+              namespace: Optional[str] = None
```

Comment:
Wait, @izellevy, this is incorrect.  
At the `KnowledgeBase` layer, each Query should still have its own individual `namespace`. The `KnowledgeBase` is used at the same "level" like the pinecone client, with the only difference that it takes data (`Document` \ `Query`) as input instead of vectors.  
Just like the same pinecone client object can handle traffic to multiple namespaces, the same `KnowledgeBase` can do the same.

Instead of deleting `namespace` from the `Query` data model, we should create a new `Query` API model (in `api_models.py`). Then, either at the `ChatEngine` layer or even the `app` layer, we should do something like:
```python
from api_models import Query 
from canopy.models.data_models import Query as InnerQuery
...
def chat(queries: List[Query], ..., namespace=None,...):
    queries = [InnerQuery(**q, namespace=namespace) for q in queries]
     ....
```

The code of `KnowledgeBase` shouldn't change at all.  
Whether the `ContextEngine` should allow namespace per query is debatable. I tend to say it does.

@miararoy WDYT?